src/python/pants/backend/native/tasks/link_shared_libraries.py

```diff
@@ -142,6 +143,16 @@ def _make_link_request(self, vt, compiled_objects_product, native_target_deps_pr
   def _get_shared_lib_cmdline_args(self):
     return self.linker.platform.resolve_platform_specific(self._SHARED_CMDLINE_ARGS)
 
+  def _get_third_party_lib_args(self):
+    lib_args = []
+    tp_files_product = self.context.products.get_data(NativeExternalLibraryFetch.NativeExternalLibraryFiles)
+    if tp_files_product.lib_names:
+      for lib_name in tp_files_product.lib_names:
+        lib_args.append('-l{}'.format(lib_name))
+      lib_dir_arg = '-L{}'.format(tp_files_product.lib)
```

Comment:
I think this logic (generating command-line arguments from `lib` and `lib_names` should be moved to the `NativeExternalLibraryFetch.NativeExternalLibraryFiles` class. That way you could remove this method entirely.

Reply:
✅